controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -721,6 +724,31 @@ private boolean isSubscriberToNonSubscriberTransition(final ReaderGroupConfigRec
            ReaderGroupState.DELETING, currentState, context, executor));
     }
 
+    /**
+     * Delete Scope Recursively.
+     *
+     * @param scope           scope to be deleted.
+     * @param requestId       request id.
+     * @return deletion status.
+     */
+    public CompletableFuture<DeleteScopeStatus.Status> deleteScopeRecursive(final String scope,
+                                                                              final long requestId) {
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        return RetryHelper.withRetriesAsync(() -> streamMetadataStore.checkScopeInDeletingTable(scope, context, executor)
+                .thenCompose(exists -> {
+                    if (exists) {
+                        log.debug(requestId, "Scope {} delete is already in progress", scope);
+                        return CompletableFuture.completedFuture(DeleteScopeStatus.Status.FAILURE);
```

Comment:
Returning a "FAILURE" status is going to indicate something went wrong on server side causing deleteScopeRecursive to fail and the Client would retry this operation. The right value to return here would be `DeleteScopeStatus.Status.SUCCESS`

Reply:
Changed as per suggestion